dipy/denoise/denspeed.pyx

```diff
@@ -108,9 +103,10 @@ def _nlmeans_3d(double[:, :, ::1] arr, double[:, :, ::1] mask,
 
     # move the block
     with nogil, parallel():
-        for i in prange(B + P, I - B - P, schedule="dynamic"):
```

Comment:
Yes looking at it. compared speeds, the `schedule = "dynamic"` makes a difference!